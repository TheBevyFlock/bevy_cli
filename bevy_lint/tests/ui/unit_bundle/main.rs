#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::unit_bundle)]

use bevy::prelude::*;
use std::f32::consts::PI;

fn main() {
    App::new().add_systems(Startup, my_system);
}

fn my_system(mut commands: Commands) {
    commands.spawn((
        Name::new("Decal"),
        Transform::from_translation(Vec3::new(0.75, 0.0, 0.0)).rotate_z(PI / 4.0),
        //~^ ERROR: created a `Bundle` containing a unit `()`
        (
            no_op(),
            //~^ ERROR: created a `Bundle` containing a unit `()`
            GlobalTransform::IDENTITY,
            (
                (),
                //~^ ERROR: created a `Bundle` containing a unit `()`
            ),
            {
                no_op();
                Transform::default()
            },
        ),
    ));
    Commands::spawn(
        &mut commands,
        (
            Name::new("Decal"),
            Transform::from_translation(Vec3::new(0.75, 0.0, 0.0)).rotate_z(PI / 4.0),
            //~^ ERROR: created a `Bundle` containing a unit `()`
            (
                no_op(),
                //~^ ERROR: created a `Bundle` containing a unit `()`
                GlobalTransform::IDENTITY,
                (
                    (),
                    //~^ ERROR: created a `Bundle` containing a unit `()`
                ),
                {
                    no_op();
                    Transform::default()
                },
            ),
        ),
    );
}

fn no_op() {}
