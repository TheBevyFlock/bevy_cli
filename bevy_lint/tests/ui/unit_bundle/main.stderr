error: created a `Bundle` containing a unit `()`
  --> tests/ui/unit_bundle/main.rs:15:9
   |
15 |         Transform::from_translation(Vec3::new(0.75, 0.0, 0.0)).rotate_z(PI / 4.0),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: units `()` are not `Component`s and will be skipped
note: the lint level is defined here
  --> tests/ui/unit_bundle/main.rs:3:9
   |
3  | #![deny(bevy::unit_bundle)]
   |         ^^^^^^^^^^^^^^^^^

error: created a `Bundle` containing a unit `()`
  --> tests/ui/unit_bundle/main.rs:18:13
   |
18 |             no_op(),
   |             ^^^^^^^
   |
   = note: units `()` are not `Component`s and will be skipped

error: created a `Bundle` containing a unit `()`
  --> tests/ui/unit_bundle/main.rs:22:17
   |
22 |                 (),
   |                 ^^
   |
   = note: units `()` are not `Component`s and will be skipped

error: created a `Bundle` containing a unit `()`
  --> tests/ui/unit_bundle/main.rs:35:13
   |
35 |             Transform::from_translation(Vec3::new(0.75, 0.0, 0.0)).rotate_z(PI / 4.0),
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: units `()` are not `Component`s and will be skipped

error: created a `Bundle` containing a unit `()`
  --> tests/ui/unit_bundle/main.rs:38:17
   |
38 |                 no_op(),
   |                 ^^^^^^^
   |
   = note: units `()` are not `Component`s and will be skipped

error: created a `Bundle` containing a unit `()`
  --> tests/ui/unit_bundle/main.rs:42:21
   |
42 |                     (),
   |                     ^^
   |
   = note: units `()` are not `Component`s and will be skipped

error: aborting due to 6 previous errors

